app/ts/tests/ethSimulate.ts

```diff
@@ -0,0 +1,60 @@
+import { EthSimulateV1Params } from './ethSimulate-types.js'
+import * as funtypes from 'funtypes'
+import { serialize } from './wire-types.js'
+
+type JsonRpcSuccessResponse = funtypes.Static<typeof JsonRpcSuccessResponse>
+const JsonRpcSuccessResponse = funtypes.ReadonlyObject({
+	jsonrpc: funtypes.Literal('2.0'),
+	id: funtypes.Union(funtypes.String, funtypes.Number),
+	result: funtypes.Unknown,
+}).asReadonly()
+
+export type JsonRpcErrorResponse = funtypes.Static<typeof JsonRpcErrorResponse>
+export const JsonRpcErrorResponse = funtypes.ReadonlyObject({
+	jsonrpc: funtypes.Literal('2.0'),
+	id: funtypes.Union(funtypes.String, funtypes.Number),
+	error: funtypes.ReadonlyObject({
+		code: funtypes.Number,
+		message: funtypes.String,
+		data: funtypes.Unknown,
+	}).asReadonly(),
+}).asReadonly()
+
+export type JsonRpcResponse = funtypes.Static<typeof JsonRpcResponse>
+export const JsonRpcResponse = funtypes.Union(JsonRpcErrorResponse, JsonRpcSuccessResponse)
+
+export type EthereumJsonRpcRequest = funtypes.Static<typeof EthereumJsonRpcRequest>
+export const EthereumJsonRpcRequest = funtypes.Union(EthSimulateV1Params)
+
+class ErrorWithData extends Error {
```

Comment:
```suggestion
export class ErrorWithData extends Error {
```

Reply:
why these need to be exported? they are not used outside the file

Reply:
For this one in particular, it seems quite likely it will be used outside of this file.  However, if you want to wait until that actually happens before you export, I won't push back on that.

Reply:
yeah, if thats used externally then its changed